Test/plot tests

jcharkow opened the pull request:
# Description

Add tests for plotting. Also add snapshots for Bokeh and Plotly plots

## Type of change
Test

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X ] This change requires a documentation update


**Test Configuration**:
* Firmware version: Ubuntu 
* Hardware: Dell XPS

# Checklist:

- [ ] My code follows the style guidelines of this project
- [X ] I have performed a self-review of my code
- [X ] I have commented my code, particularly in hard-to-understand areas
- [X ] I have made corresponding changes to the documentation
- [X ] My changes generate no new warnings
- [X ] I have added tests that prove my fix is effective or that my feature works
- [X ] New and existing unit tests pass locally with my changes
- [ X] Any dependent changes have been merged and published in downstream modules
<!--- START AUTOGENERATED NOTES --->
# Contents ([#119](https://github.com/Roestlab/massdash/pull/119))

### Other
- test: clean up imports update API docs
- update plotly snapshot
- check approximate equality with floats
- fix ployly snapshot test?
- update bokeh snapshot
- make plotly snapshot more leniant
- relax numerical inaccuracy more
- remove surface plot im vs m/z test from win32
- gui inconsitencies
- test: update sigma
- update readme for testing

### Uncategorised!
- [FEATURE] add BokehSnapShot
- [FIX][TEST] bugs in bokeh json snapshot comparisons
- [TEST] InteractivePlotter1D tests
- [TEST] add test for plot function
- move rgb_to_hex() function to util
- code lint
- [TEST] create ionMobility Feature DF for testing
- [TEST][FIX] update InteractivePlotter test
- [TEST] interactive2D plotter test
- [TEST] remove unused code
- [TEST] add tests for 3D plotter
- [TEST] add tests for plotConfig
- add file headers to test files
- Merge branch 'dev' into test/plot_tests
- Merge branch 'dev' into test/plot_tests

<!--- END AUTOGENERATED NOTES --->

jcharkow committed:
[FEATURE] add BokehSnapShot

Add BokehSnapshot class to compare the outputs of bokeh plots.

Bokeh plots are saved as html files so that they can be opened and
examined in the browser. Only the "json" section is compared and the id
tags are ignored.

jcharkow committed:
[FIX][TEST] bugs in bokeh json snapshot comparisons

Previously not accounting for different ordering of attributes in bokeh
json, now account for different ordering.

jcharkow committed:
[TEST] InteractivePlotter1D tests

Tests for interactive plotter 1D

jcharkow committed:
[TEST] add test for plot function

jcharkow committed:
move rgb_to_hex() function to util

jcharkow committed:
code lint

jcharkow committed:
[TEST] create ionMobility Feature DF for testing

based on ionMobility.mzML file, save a df of featureMap for testing 2d
plotting

jcharkow committed:
[TEST][FIX] update InteractivePlotter test

update test so that gaussian smoothing is actually applied

jcharkow committed:
[TEST] interactive2D plotter test

tests for interactive 2D plotter, also minor bug fix for interactive
plotter

jcharkow committed:
[TEST] remove unused code

jcharkow committed:
[TEST] add tests for 3D plotter

minor update to 3d plotter so throw proper error if no plot selected for
surface plots

add plotly snapshot

NOTE: some 3d plotting functions are borken and therefore not tested

jcharkow committed:
[TEST] add tests for plotConfig

jcharkow committed:
add file headers to test files

jcharkow committed:
Merge branch 'dev' into test/plot_tests

jcharkow committed:
doc: test: clean up imports update API docs

update API docs and clean up import of Plot snapshots

jcharkow committed:
test: update plotly snapshot

plotly snapshot were incorrect, update to actually use json format

jcharkow committed:
test: check approximate equality with floats

jcharkow committed:
test: fix ployly snapshot test?

jcharkow committed:
test: update bokeh snapshot

update bokeh snapshot to be more verbose and more resistant to changes

jcharkow committed:
test: make plotly snapshot more leniant

make snapshot more leniant so pass on windows

jcharkow committed:
test: relax numerical inaccuracy more

jcharkow committed:
test:remove surface plot im vs m/z test from win32

this test fails on windows, seems plot is different so remove for now.

singjc reviewed (approved):
Great job! A lot of new tests! :partying_face: 

1. Just make sure the GUI works with the small change made to some of plotting (i.e. smoothing), just to make sure the GUI still works as expected. If you already tested and verified, then that's great!

2. Can you update the tests README to document the change in the different unit testing strategy using syrup and pytest, and add a small example for how one would implement a test for a new method.

Good to merge otherwise, you can merge when ready.

jcharkow committed:
Merge branch 'dev' into test/plot_tests

jcharkow committed:
fix: gui inconsitencies

rename sigma parameter to gaussian sigma to be consistent with gui (and
avoid crash)

jcharkow committed:
fix: test: update sigma

update test plotConfig from sigma to gaussian_sigma

jcharkow committed:
doc: update readme for testing

Also fix badge on main docs page
